src/deepqmc/wf/paulinet/paulinet.py

```diff
@@ -173,25 +153,18 @@ def __init__(
             backflow_spec[1] *= 2
         self.backflow_type = backflow_type
         self.backflow_transform = backflow_transform
-        self.backflow = (
-            backflow_factory(len(mol), dist_feat_dim, n_up, n_down, *backflow_spec)
-            if backflow_factory
+        self.omni = (
+            omni_factory(mol, n_up, n_down, backflow_spec, **omni_kwargs)
+            if omni_factory
             else None
         )
-        self.r_backflow = None
-        if omni_factory:
-            assert not backflow_factory and not jastrow_factory
-            self.omni = omni_factory(
-                mol, dist_feat_dim, n_up, n_down, *backflow_spec, **(omni_kwargs or {})
-            )
-            self.backflow = self.omni.forward_backflow
-            self.r_backflow = self.omni.forward_r_backflow
-            self.jastrow = self.omni.forward_jastrow
-        else:
-            self.omni = None
+        self.jastrow = (
```

Comment:
I'm thinking whether we shouldn't simplify this, now that `jastrow` and `backflow` will always refer to the functions from `omni`. Not sure

Reply:
I think everything would simplify a lot, if we had a single `forward()` on omni, that would return both the jastrow and the backflow. Then this whole business with the "memory" can be removed, and the stream logic will be much simpler. What do you think?

Reply:
You are right, this is not needed and both could be combined in a single forward call. I thought that this way its more tracable what happens and the forward of the PauliNet does not have to be adapted, though I'd be open for changing it. However I do not see how that would allow the memory to be removed and how it would change the stream logic. Consider the following example: Both Jastrow and backflow are build upon the many-body embeddings. Then I would still have to store the many-body embeddings after the first forward call of the ElectronicSchNet for lets say the Jastrow, to be reused in the backflow. Or another example: Jastrow uses many-body embeddings and backflow uses edges. Then I would like to store the edges in the process of obtaining the many-body representation to be reused when called in the backflow. Did I miss something?

Reply:
Speaking of _the first forward call of the ElectronicSchNet for lets say the **Jastrow**, to be reused in the **backflow**_ I mean Jastrow and backflow calles in omni.py.